// blizzard.js/dist/resources/d3.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const act = (args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'd3/data/act' : `d3/data/act/${encodeURIComponent(args.id)}`,
    };
};
const artisan = (args) => {
    return {
        path: `d3/data/artisan/${encodeURIComponent(args.id)}`,
    };
};
const recipe = (args) => {
    return {
        path: `d3/data/artisan/${encodeURIComponent(args.artisan)}/recipe/${encodeURIComponent(args.recipe)}`,
    };
};
const follower = (args) => {
    return {
        path: `d3/data/follower/${encodeURIComponent(args.id)}`,
    };
};
const characterClass = (args) => {
    return {
        path: `d3/data/hero/${encodeURIComponent(args.id)}`,
    };
};
const characterSkill = (args) => {
    return {
        path: `d3/data/hero/${encodeURIComponent(args.class)}/skill/${encodeURIComponent(args.skill)}`,
    };
};
const item = (args) => {
    return {
        path: `d3/data/item/${encodeURIComponent(args.id)}`,
    };
};
const itemType = (args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'd3/data/item-type' : `d3/data/item-type/${encodeURIComponent(args.id)}`,
    };
};
const accountProfile = (args) => {
    if (args.hero === undefined) {
        return {
            path: `d3/profile/${encodeURIComponent(args.account)}/`,
        };
    }
    return {
        path: args.resource === undefined
            ? `d3/profile/${encodeURIComponent(args.account)}/hero/${encodeURIComponent(args.hero)}`
            : `d3/profile/${encodeURIComponent(args.account)}/hero/${encodeURIComponent(args.hero)}/${encodeURIComponent(args.resource)}`,
    };
};
const season = (args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/d3/season/',
        };
    }
    return {
        path: args.leaderboard === undefined
            ? `data/d3/season/${encodeURIComponent(args.id)}`
            : `data/d3/season/${encodeURIComponent(args.id)}/leaderboard/${encodeURIComponent(args.leaderboard)}`,
    };
};
const era = (args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/d3/era/',
        };
    }
    return {
        path: args.leaderboard === undefined
            ? `data/d3/era/${encodeURIComponent(args.id)}`
            : `data/d3/era/${encodeURIComponent(args.id)}/leaderboard/${encodeURIComponent(args.leaderboard)}`,
    };
};

exports.accountProfile = accountProfile;
exports.act = act;
exports.artisan = artisan;
exports.characterClass = characterClass;
exports.characterSkill = characterSkill;
exports.era = era;
exports.follower = follower;
exports.item = item;
exports.itemType = itemType;
exports.recipe = recipe;
exports.season = season;
