// blizzard.js/dist/resources/hs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const cardSearch = (args) => {
    return {
        path: 'hearthstone/cards',
        params: {
            attack: Array.isArray(args.attack) ? args.attack.join(',') : args.attack,
            class: args.class,
            collectible: args.collectible === undefined ? '0,1' : args.collectible,
            gameMode: args.tier === undefined ? args.gameMode : 'battlegrounds',
            health: Array.isArray(args.health) ? args.health.join(',') : args.health,
            keyword: args.keyword,
            manaCost: Array.isArray(args.manaCost) ? args.manaCost.join(',') : args.manaCost,
            minionType: args.minionType,
            order: args.order,
            page: args.page,
            pageSize: args.pageSize,
            rarity: args.rarity,
            set: args.set,
            sort: Array.isArray(args.sort) ? args.sort.join(',') : args.sort,
            textFilter: args.textFilter,
            tier: Array.isArray(args.tier) ? args.tier.join(',') : args.tier,
            type: args.type,
        },
    };
};
const card = (args) => {
    return {
        path: `hearthstone/cards/${encodeURIComponent(args.id)}`,
        params: {
            gameMode: args.gameMode,
        },
    };
};
const cardBacks = (args) => {
    if (args.id !== undefined) {
        return {
            path: `hearthstone/cardbacks/${encodeURIComponent(args.id)}`,
        };
    }
    return {
        path: 'hearthstone/cardbacks',
        params: {
            cardBackCategory: args.category,
            order: args.order,
            sort: args.sort,
            textFilter: args.textFilter,
        },
    };
};
const deck = (args) => {
    return {
        path: 'hearthstone/deck',
        params: {
            code: args.code,
            ids: Array.isArray(args.ids) ? args.ids.join(',') : args.ids,
            hero: args.hero,
        },
    };
};
const metadata = (args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.type) === undefined ? 'hearthstone/metadata' : `hearthstone/metadata/${encodeURIComponent(args.type)}`,
    };
};

exports.card = card;
exports.cardBacks = cardBacks;
exports.cardSearch = cardSearch;
exports.deck = deck;
exports.metadata = metadata;
