// blizzard.js/dist/resources/sc2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const profile = (args) => {
    return {
        path: args.realm === undefined || args.profile === undefined
            ? `sc2/static/profile/${encodeURIComponent(args.region)}`
            : `sc2/static/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}`,
    };
};
const metadata = (args) => {
    return {
        path: `sc2/metadata/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}`,
    };
};
const profileLadder = (args) => {
    return {
        path: args.id === undefined
            ? `sc2/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}/ladder/summary`
            : `sc2/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}/ladder/${encodeURIComponent(args.id)}`,
    };
};
const ladder = (args) => {
    return {
        path: `sc2/ladder/${encodeURIComponent(args.resource)}/${encodeURIComponent(args.region)}`,
    };
};
const account = (args) => {
    return {
        path: `sc2/player/${encodeURIComponent(args.id)}`,
    };
};
const legacyProfile = (args) => {
    return {
        path: args.resource === undefined
            ? `sc2/legacy/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}`
            : `sc2/legacy/profile/${encodeURIComponent(args.region)}/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.profile)}/${encodeURIComponent(args.resource)}`,
    };
};
const legacyLadder = (args) => {
    return {
        path: `sc2/legacy/ladder/${encodeURIComponent(args.region)}/${encodeURIComponent(args.ladder)}`,
    };
};
const legacyAchievements = (args) => {
    return {
        path: `sc2/legacy/achievements/${encodeURIComponent(args.region)}`,
    };
};
const legacyRewards = (args) => {
    return {
        path: `sc2/legacy/rewards/${encodeURIComponent(args.region)}`,
    };
};
const league = (args) => {
    return {
        path: `data/sc2/league/${encodeURIComponent(args.season)}/${encodeURIComponent(args.queue)}/${encodeURIComponent(args.teamType)}/${encodeURIComponent(args.league)}`,
    };
};

exports.account = account;
exports.ladder = ladder;
exports.league = league;
exports.legacyAchievements = legacyAchievements;
exports.legacyLadder = legacyLadder;
exports.legacyProfile = legacyProfile;
exports.legacyRewards = legacyRewards;
exports.metadata = metadata;
exports.profile = profile;
exports.profileLadder = profileLadder;
