// blizzard.js/dist/resources/wow.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const accountProfile = (namespace) => {
    return {
        path: 'profile/user/wow',
        namespace,
    };
};
const accountCharacterProfile = (namespace, args) => {
    return {
        path: `profile/user/wow/protected-character/${encodeURIComponent(args.realm)}-${encodeURIComponent(args.character)}`,
        namespace,
    };
};
const accountCollections = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.resource) === undefined
            ? 'profile/user/wow/collections'
            : `profile/user/wow/collections/${encodeURIComponent(args.resource)}`,
        namespace,
    };
};
const characterAchievements = (namespace, args) => {
    return {
        path: args.stats
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/achievements/statistics`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/achievements`,
        namespace,
    };
};
const characterAppearance = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/appearance`,
        namespace,
    };
};
const characterCollections = (namespace, args) => {
    return {
        path: args.resource === undefined
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/collections`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/collections/${encodeURIComponent(args.resource)}`,
        namespace,
    };
};
const characterEncounters = (namespace, args) => {
    return {
        path: args.resource === undefined
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/encounters`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/encounters/${encodeURIComponent(args.resource)}`,
        namespace,
    };
};
const characterEquipment = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/equipment`,
        namespace,
    };
};
const characterHunterPets = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/hunter-pets`,
        namespace,
    };
};
const characterMedia = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/character-media`,
        namespace,
    };
};
const characterMythicKeystone = (namespace, args) => {
    return {
        path: args.season === undefined
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/mythic-keystone-profile`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/mythic-keystone-profile/season/${encodeURIComponent(args.season)}`,
        namespace,
    };
};
const characterProfessions = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/professions`,
        namespace,
    };
};
const characterProfile = (namespace, args) => {
    return {
        path: args.status
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/status`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}`,
        namespace,
    };
};
const characterPVP = (namespace, args) => {
    return {
        path: args.bracket === undefined
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/pvp-summary`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/pvp-bracket/${encodeURIComponent(args.bracket)}`,
        namespace,
    };
};
const characterQuests = (namespace, args) => {
    return {
        path: args.completed
            ? `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/quests/completed`
            : `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/quests`,
        namespace,
    };
};
const characterReputations = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/reputations`,
        namespace,
    };
};
const characterSoulbinds = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/soulbinds`,
        namespace,
    };
};
const characterSpecializations = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/specializations`,
        namespace,
    };
};
const characterStatistics = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/statistics`,
        namespace,
    };
};
const characterTitles = (namespace, args) => {
    return {
        path: `profile/wow/character/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/titles`,
        namespace,
    };
};
const guild = (namespace, args) => {
    return {
        path: args.resource !== 'activity' && args.resource !== 'achievements' && args.resource !== 'roster'
            ? `data/wow/guild/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}`
            : `data/wow/guild/${encodeURIComponent(args.realm)}/${encodeURIComponent(args.name)}/${encodeURIComponent(args.resource)}`,
        namespace,
    };
};
const achievement = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/achievement/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/achievement/${encodeURIComponent(args.id)}`
            : `data/wow/achievement/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const achievementCategory = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/achievement-category/index',
            namespace,
        };
    }
    return {
        path: `data/wow/achievement-category/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const auctionHouse = (namespace, args) => {
    return {
        path: `data/wow/connected-realm/${encodeURIComponent(args.id)}/auctions`,
        namespace,
    };
};
const azeriteEssence = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/azerite-essence/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/azerite-essence/${encodeURIComponent(args.id)}`
            : `data/wow/azerite-essence/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const azeriteEssenceSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/azerite-essence',
        namespace,
        params: {
            'allowed_specializations.id': args.id,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const commodities = (namespace) => {
    return {
        path: `data/wow/auctions/commodities`,
        namespace,
    };
};
const connectedRealm = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/connected-realm/index',
            namespace,
        };
    }
    return {
        path: `data/wow/connected-realm/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const connectedRealmSearch = (namespace, args) => {
    return {
        path: `data/wow/search/connected-realm`,
        namespace,
        params: {
            'status.type': args.status,
            'realms.timezone': args.timezone,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const covenant = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/covenant/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/covenant/${encodeURIComponent(args.id)}`
            : `data/wow/covenant/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const soulbind = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/soulbind/index',
            namespace,
        };
    }
    return {
        path: `data/wow/soulbind/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const conduit = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/covenant/conduit/index',
            namespace,
        };
    }
    return {
        path: `data/wow/covenant/conduit/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const creature = (namespace, args) => {
    return {
        path: args.media
            ? `data/wow/media/creature-display/${encodeURIComponent(args.id)}`
            : `data/wow/creature/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const creatureFamily = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/creature-family/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/creature-family/${encodeURIComponent(args.id)}`
            : `data/wow/creature-family/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const creatureType = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/creature-type/index',
            namespace,
        };
    }
    return {
        path: `data/wow/creature-type/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const creatureSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/creature',
        namespace,
        params: {
            [`name.${args.locale}`]: args.name,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const guildCrest = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.resource) === undefined && (args === null || args === void 0 ? void 0 : args.id) == undefined
            ? `data/wow/guild-crest/index`
            : `data/wow/media/guild-crest/${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const item = (namespace, args) => {
    if (args.resource === 'set') {
        return {
            path: args.id === undefined ? 'data/wow/item-set/index' : `data/wow/item-set/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    if (args.resource === 'class') {
        if (args.id === undefined) {
            return {
                path: 'data/wow/item-class/index',
                namespace,
            };
        }
        return {
            path: args.sub === undefined
                ? `data/wow/item-class/${encodeURIComponent(args.id)}`
                : `data/wow/item-class/${encodeURIComponent(args.id)}/item-subclass/${encodeURIComponent(args.sub)}`,
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/item/${encodeURIComponent(args.id)}`
            : `data/wow/item/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const itemSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/item',
        namespace,
        params: {
            [`name.${args.locale}`]: args.name,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const journal = (namespace, args) => {
    if (args.id === undefined) {
        return {
            path: `data/wow/journal-${encodeURIComponent(args.resource)}/index`,
            namespace,
        };
    }
    return {
        path: args.resource === 'instance' && args.media
            ? `data/wow/media/journal-${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`
            : `data/wow/journal-${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const mediaSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/media',
        namespace,
        params: {
            _tag: args.tag,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const modifiedCrafting = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.resource) === 'category') {
        return {
            path: (args === null || args === void 0 ? void 0 : args.id) === undefined
                ? 'data/wow/modified-crafting/category/index'
                : `data/wow/modified-crafting/category/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    if ((args === null || args === void 0 ? void 0 : args.resource) === 'reagent') {
        return {
            path: (args === null || args === void 0 ? void 0 : args.id) === undefined
                ? 'data/wow/modified-crafting/reagent-slot-type/index'
                : `data/wow/modified-crafting/reagent-slot-type/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    return {
        path: 'data/wow/modified-crafting/index',
        namespace,
    };
};
const mount = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/mount/index',
            namespace,
        };
    }
    return {
        path: `data/wow/mount/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const mountSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/mount',
        namespace,
        params: {
            [`name.${args.locale}`]: args.name,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const mythicKeystone = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.resource) !== 'dungeon' && (args === null || args === void 0 ? void 0 : args.resource) !== 'period' && (args === null || args === void 0 ? void 0 : args.resource) !== 'season') {
        return {
            path: 'data/wow/mythic-keystone/index',
            namespace,
        };
    }
    return {
        path: args.id === undefined
            ? `data/wow/mythic-keystone/${encodeURIComponent(args.resource)}/index`
            : `data/wow/mythic-keystone/${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const mythicKeystoneAffix = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/keystone-affix/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/keystone-affix/${encodeURIComponent(args.id)}`
            : `data/wow/keystone-affix/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const mythicKeystoneLeaderboard = (namespace, args) => {
    return {
        path: args.dungeon === undefined || args.period === undefined
            ? `data/wow/connected-realm/${encodeURIComponent(args.realm)}/mythic-leaderboard/index`
            : `data/wow/connected-realm/${encodeURIComponent(args.realm)}/mythic-leaderboard/${encodeURIComponent(args.dungeon)}/period/${encodeURIComponent(args.period)}`,
        namespace,
    };
};
const mythicRaidLeaderboard = (namespace, args) => {
    return {
        path: `data/wow/leaderboard/hall-of-fame/${encodeURIComponent(args.raid)}/${encodeURIComponent(args.faction)}`,
        namespace,
    };
};
const pet = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.resource) === 'ability') {
        if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
            return {
                path: 'data/wow/pet-ability/index',
                namespace,
            };
        }
        return {
            path: (args === null || args === void 0 ? void 0 : args.media)
                ? `data/wow/media/pet-ability/${encodeURIComponent(args.id)}`
                : `data/wow/pet-ability/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/pet/index',
            namespace,
        };
    }
    return {
        path: (args === null || args === void 0 ? void 0 : args.media)
            ? `data/wow/media/pet/${encodeURIComponent(args.id)}`
            : `data/wow/pet/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const playableClass = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: `data/wow/playable-class/index`,
            namespace,
        };
    }
    if (args.pvpTalents) {
        return {
            path: `data/wow/playable-class/${encodeURIComponent(args.id)}/pvp-talent-slots`,
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/playable-class/${encodeURIComponent(args.id)}`
            : `data/wow/playable-class/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const playableRace = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'data/wow/playable-race/index' : `data/wow/playable-race/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const playableSpecialization = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/playable-specialization/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/playable-specialization/${encodeURIComponent(args.id)}`
            : `data/wow/playable-specialization/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const powerType = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'data/wow/power-type/index' : `data/wow/power-type/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const profession = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/profession/index',
            namespace,
        };
    }
    if (args.skillTier !== undefined) {
        return {
            path: `data/wow/profession/${encodeURIComponent(args.id)}/skill-tier/${encodeURIComponent(args.skillTier)}`,
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/profession/${encodeURIComponent(args.id)}`
            : `data/wow/profession/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const recipe = (namespace, args) => {
    return {
        path: args.media
            ? `data/wow/media/recipe/${encodeURIComponent(args.id)}`
            : `data/wow/recipe/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const pvpSeason = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/pvp-season/index',
            namespace,
        };
    }
    if (args.resource === 'leaderboard') {
        return {
            path: args.bracket === undefined
                ? `data/wow/pvp-season/${encodeURIComponent(args.id)}/pvp-leaderboard/index`
                : `data/wow/pvp-season/${encodeURIComponent(args.id)}/pvp-leaderboard/${encodeURIComponent(args.bracket)}`,
            namespace,
        };
    }
    if (args.resource === 'reward') {
        return {
            path: `data/wow/pvp-season/${encodeURIComponent(args.id)}/pvp-reward/index`,
            namespace,
        };
    }
    return {
        path: `data/wow/pvp-season/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const pvpTier = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/pvp-tier/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/pvp-tier/${encodeURIComponent(args.id)}`
            : `data/wow/pvp-tier/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const quest = (namespace, args) => {
    if ((args === null || args === void 0 ? void 0 : args.resource) === 'category' || (args === null || args === void 0 ? void 0 : args.resource) === 'area' || (args === null || args === void 0 ? void 0 : args.resource) === 'type') {
        return {
            path: args.id === undefined
                ? `data/wow/quest/${encodeURIComponent(args.resource)}/index`
                : `data/wow/quest/${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/quest/index',
            namespace,
        };
    }
    return {
        path: `data/wow/quest/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const realm = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.slug) === undefined ? 'data/wow/realm/index' : `data/wow/realm/${encodeURIComponent(args.slug)}`,
        namespace,
    };
};
const realmSearch = (namespace, args) => {
    return {
        path: 'data/wow/search/realm',
        namespace,
        params: {
            timezone: args === null || args === void 0 ? void 0 : args.timezone,
            orderby: Array.isArray(args === null || args === void 0 ? void 0 : args.orderby) ? args.orderby.join(',') : args === null || args === void 0 ? void 0 : args.orderby,
            _page: args.page,
        },
    };
};
const region = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'data/wow/region/index' : `data/wow/region/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const reputation = (namespace, args) => {
    return {
        path: args.id === undefined
            ? `data/wow/reputation-${encodeURIComponent(args.resource)}/index`
            : `data/wow/reputation-${encodeURIComponent(args.resource)}/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const spell = (namespace, args) => {
    return {
        path: args.media
            ? `data/wow/media/spell/${encodeURIComponent(args.id)}`
            : `data/wow/spell/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const spellSearch = (namespace, args) => {
    return {
        path: `data/wow/search/spell`,
        namespace,
        params: {
            [`name.${args.locale}`]: args.name,
            orderby: Array.isArray(args.orderby) ? args.orderby.join(',') : args.orderby,
            _page: args.page,
        },
    };
};
const talent = (namespace, args) => {
    if (args === null || args === void 0 ? void 0 : args.pvp) {
        return {
            path: args.id === undefined ? 'data/wow/pvp-talent/index' : `data/wow/pvp-talent/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'data/wow/talent/index' : `data/wow/talent/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const techTalent = (namespace, args) => {
    if (args === null || args === void 0 ? void 0 : args.tree) {
        return {
            path: args.id === undefined
                ? 'data/wow/tech-talent-tree/index'
                : `data/wow/tech-talent-tree/${encodeURIComponent(args.id)}`,
            namespace,
        };
    }
    if ((args === null || args === void 0 ? void 0 : args.id) === undefined) {
        return {
            path: 'data/wow/tech-talent/index',
            namespace,
        };
    }
    return {
        path: args.media
            ? `data/wow/media/tech-talent/${encodeURIComponent(args.id)}`
            : `data/wow/tech-talent/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const title = (namespace, args) => {
    return {
        path: (args === null || args === void 0 ? void 0 : args.id) === undefined ? 'data/wow/title/index' : `data/wow/title/${encodeURIComponent(args.id)}`,
        namespace,
    };
};
const token = (namespace) => {
    return {
        path: 'data/wow/token/index',
        namespace,
    };
};

exports.accountCharacterProfile = accountCharacterProfile;
exports.accountCollections = accountCollections;
exports.accountProfile = accountProfile;
exports.achievement = achievement;
exports.achievementCategory = achievementCategory;
exports.auctionHouse = auctionHouse;
exports.azeriteEssence = azeriteEssence;
exports.azeriteEssenceSearch = azeriteEssenceSearch;
exports.characterAchievements = characterAchievements;
exports.characterAppearance = characterAppearance;
exports.characterCollections = characterCollections;
exports.characterEncounters = characterEncounters;
exports.characterEquipment = characterEquipment;
exports.characterHunterPets = characterHunterPets;
exports.characterMedia = characterMedia;
exports.characterMythicKeystone = characterMythicKeystone;
exports.characterPVP = characterPVP;
exports.characterProfessions = characterProfessions;
exports.characterProfile = characterProfile;
exports.characterQuests = characterQuests;
exports.characterReputations = characterReputations;
exports.characterSoulbinds = characterSoulbinds;
exports.characterSpecializations = characterSpecializations;
exports.characterStatistics = characterStatistics;
exports.characterTitles = characterTitles;
exports.commodities = commodities;
exports.conduit = conduit;
exports.connectedRealm = connectedRealm;
exports.connectedRealmSearch = connectedRealmSearch;
exports.covenant = covenant;
exports.creature = creature;
exports.creatureFamily = creatureFamily;
exports.creatureSearch = creatureSearch;
exports.creatureType = creatureType;
exports.guild = guild;
exports.guildCrest = guildCrest;
exports.item = item;
exports.itemSearch = itemSearch;
exports.journal = journal;
exports.mediaSearch = mediaSearch;
exports.modifiedCrafting = modifiedCrafting;
exports.mount = mount;
exports.mountSearch = mountSearch;
exports.mythicKeystone = mythicKeystone;
exports.mythicKeystoneAffix = mythicKeystoneAffix;
exports.mythicKeystoneLeaderboard = mythicKeystoneLeaderboard;
exports.mythicRaidLeaderboard = mythicRaidLeaderboard;
exports.pet = pet;
exports.playableClass = playableClass;
exports.playableRace = playableRace;
exports.playableSpecialization = playableSpecialization;
exports.powerType = powerType;
exports.profession = profession;
exports.pvpSeason = pvpSeason;
exports.pvpTier = pvpTier;
exports.quest = quest;
exports.realm = realm;
exports.realmSearch = realmSearch;
exports.recipe = recipe;
exports.region = region;
exports.reputation = reputation;
exports.soulbind = soulbind;
exports.spell = spell;
exports.spellSearch = spellSearch;
exports.talent = talent;
exports.techTalent = techTalent;
exports.title = title;
exports.token = token;
