// blizzard.js/dist/endpoints.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const endpoints = {
    us: {
        hostname: 'https://us.api.blizzard.com',
        defaultLocale: 'en_US',
        locales: ['en_US', 'es_MX', 'pt_BR', 'multi'],
    },
    eu: {
        hostname: 'https://eu.api.blizzard.com',
        defaultLocale: 'en_GB',
        locales: ['en_GB', 'es_ES', 'fr_FR', 'ru_RU', 'de_DE', 'pt_PT', 'it_IT', 'multi'],
    },
    sea: {
        hostname: 'https://sea.api.blizzard.com',
        defaultLocale: 'en_US',
        locales: ['en_US', 'multi'],
    },
    kr: {
        hostname: 'https://kr.api.blizzard.com',
        defaultLocale: 'ko_KR',
        locales: ['ko_KR', 'en_GB', 'en_US', 'multi'],
    },
    tw: {
        hostname: 'https://tw.api.blizzard.com',
        defaultLocale: 'zh_TW',
        locales: ['zh_TW', 'en_GB', 'en_US', 'multi'],
    },
};
function getEndpoint(origin = 'us', locale) {
    var _a;
    const endpoint = endpoints[origin];
    return {
        origin,
        hostname: endpoint.hostname,
        locale: (_a = endpoint.locales.find((item) => item === locale)) !== null && _a !== void 0 ? _a : endpoint.defaultLocale,
    };
}

exports.getEndpoint = getEndpoint;
