// blizzard.js/dist/sc2/client.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('../core/core.js');
var sc2 = require('../resources/sc2.js');

class SC2 extends core.Blizzard {
    constructor() {
        super(...arguments);
        this.profile = this.createClientResourceRequest(sc2.profile);
        this.metadata = this.createClientResourceRequest(sc2.metadata);
        this.profileLadder = this.createClientResourceRequest(sc2.profileLadder);
        this.ladder = this.createClientResourceRequest(sc2.ladder);
        this.account = this.createClientResourceRequest(sc2.account);
        this.legacyProfile = this.createClientResourceRequest(sc2.legacyProfile);
        this.legacyLadder = this.createClientResourceRequest(sc2.legacyLadder);
        this.legacyAchievements = this.createClientResourceRequest(sc2.legacyAchievements);
        this.legacyRewards = this.createClientResourceRequest(sc2.legacyRewards);
        this.league = this.createClientResourceRequest(sc2.league);
    }
}

exports.SC2 = SC2;
