// blizzard.js/dist/wow/classic/client.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('../../core/core.js');
var classic = require('../../resources/classic.js');
var wow = require('../../resources/wow.js');

class WoWClassic extends core.Blizzard {
    constructor() {
        super(...arguments);
        this.auctionHouse = this.createClientResourceRequest(classic.auctionHouse.bind(this, 'dynamic-classic'));
        this.connectedRealm = this.createClientResourceRequest(wow.connectedRealm.bind(this, 'dynamic-classic'));
        this.connectedRealmSearch = this.createClientResourceRequest(wow.connectedRealmSearch.bind(this, 'dynamic-classic'));
        this.creature = this.createClientResourceRequest(wow.creature.bind(this, 'static-classic'));
        this.creatureFamily = this.createClientResourceRequest(wow.creatureFamily.bind(this, 'static-classic'));
        this.creatureSearch = this.createClientResourceRequest(wow.creatureSearch.bind(this, 'static-classic'));
        this.creatureType = this.createClientResourceRequest(wow.creatureType.bind(this, 'static-classic'));
        this.guildCrest = this.createClientResourceRequest(wow.guildCrest.bind(this, 'static-classic'));
        this.item = this.createClientResourceRequest(wow.item.bind(this, 'static-classic'));
        this.itemSearch = this.createClientResourceRequest(wow.itemSearch.bind(this, 'static-classic'));
        this.mediaSearch = this.createClientResourceRequest(wow.mediaSearch.bind(this, 'static-classic'));
        this.playableClass = this.createClientResourceRequest(wow.playableClass.bind(this, 'static-classic'));
        this.playableRace = this.createClientResourceRequest(wow.playableRace.bind(this, 'static-classic'));
        this.powerType = this.createClientResourceRequest(wow.powerType.bind(this, 'static-classic'));
        this.realm = this.createClientResourceRequest(wow.realm.bind(this, 'dynamic-classic'));
        this.realmSearch = this.createClientResourceRequest(wow.realmSearch.bind(this, 'dynamic-classic'));
        this.region = this.createClientResourceRequest(wow.region.bind(this, 'dynamic-classic'));
        this.token = this.createClientResourceRequest(wow.token.bind(this, 'dynamic-classic'));
    }
}

exports.WoWClassic = WoWClassic;
