// blizzard.js/dist/wow/client.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('../core/core.js');
var wow = require('../resources/wow.js');

class WoW extends core.Blizzard {
    constructor() {
        super(...arguments);
        this.accountCharacterProfile = this.createClientResourceRequest(wow.accountCharacterProfile.bind(this, 'profile'));
        this.accountCollections = this.createClientResourceRequest(wow.accountCollections.bind(this, 'profile'));
        this.accountProfile = this.createClientResourceRequest(wow.accountProfile.bind(this, 'profile'));
        this.achievement = this.createClientResourceRequest(wow.achievement.bind(this, 'static'));
        this.achievementCategory = this.createClientResourceRequest(wow.achievementCategory.bind(this, 'static'));
        this.auctionHouse = this.createClientResourceRequest(wow.auctionHouse.bind(this, 'dynamic'));
        this.azeriteEssence = this.createClientResourceRequest(wow.azeriteEssence.bind(this, 'static'));
        this.azeriteEssenceSearch = this.createClientResourceRequest(wow.azeriteEssenceSearch.bind(this, 'static'));
        this.characterAchievements = this.createClientResourceRequest(wow.characterAchievements.bind(this, 'profile'));
        this.characterAppearance = this.createClientResourceRequest(wow.characterAppearance.bind(this, 'profile'));
        this.characterCollections = this.createClientResourceRequest(wow.characterCollections.bind(this, 'profile'));
        this.characterEncounters = this.createClientResourceRequest(wow.characterEncounters.bind(this, 'profile'));
        this.characterEquipment = this.createClientResourceRequest(wow.characterEquipment.bind(this, 'profile'));
        this.characterHunterPets = this.createClientResourceRequest(wow.characterHunterPets.bind(this, 'profile'));
        this.characterMedia = this.createClientResourceRequest(wow.characterMedia.bind(this, 'profile'));
        this.characterMythicKeystone = this.createClientResourceRequest(wow.characterMythicKeystone.bind(this, 'profile'));
        this.characterProfessions = this.createClientResourceRequest(wow.characterProfessions.bind(this, 'profile'));
        this.characterProfile = this.createClientResourceRequest(wow.characterProfile.bind(this, 'profile'));
        this.characterPVP = this.createClientResourceRequest(wow.characterPVP.bind(this, 'profile'));
        this.characterQuests = this.createClientResourceRequest(wow.characterQuests.bind(this, 'profile'));
        this.characterReputations = this.createClientResourceRequest(wow.characterReputations.bind(this, 'profile'));
        this.characterSoulbinds = this.createClientResourceRequest(wow.characterSoulbinds.bind(this, 'profile'));
        this.characterSpecializations = this.createClientResourceRequest(wow.characterSpecializations.bind(this, 'profile'));
        this.characterStatistics = this.createClientResourceRequest(wow.characterStatistics.bind(this, 'profile'));
        this.characterTitles = this.createClientResourceRequest(wow.characterTitles.bind(this, 'profile'));
        this.conduit = this.createClientResourceRequest(wow.conduit.bind(this, 'static'));
        this.commodities = this.createClientResourceRequest(wow.commodities.bind(this, 'dynamic'));
        this.connectedRealm = this.createClientResourceRequest(wow.connectedRealm.bind(this, 'dynamic'));
        this.connectedRealmSearch = this.createClientResourceRequest(wow.connectedRealmSearch.bind(this, 'dynamic'));
        this.covenant = this.createClientResourceRequest(wow.covenant.bind(this, 'static'));
        this.creature = this.createClientResourceRequest(wow.creature.bind(this, 'static'));
        this.creatureFamily = this.createClientResourceRequest(wow.creatureFamily.bind(this, 'static'));
        this.creatureSearch = this.createClientResourceRequest(wow.creatureSearch.bind(this, 'static'));
        this.creatureType = this.createClientResourceRequest(wow.creatureType.bind(this, 'static'));
        this.guild = this.createClientResourceRequest(wow.guild.bind(this, 'profile'));
        this.guildCrest = this.createClientResourceRequest(wow.guildCrest.bind(this, 'static'));
        this.item = this.createClientResourceRequest(wow.item.bind(this, 'static'));
        this.itemSearch = this.createClientResourceRequest(wow.itemSearch.bind(this, 'static'));
        this.journal = this.createClientResourceRequest(wow.journal.bind(this, 'static'));
        this.mediaSearch = this.createClientResourceRequest(wow.mediaSearch.bind(this, 'static'));
        this.modifiedCrafting = this.createClientResourceRequest(wow.modifiedCrafting.bind(this, 'static'));
        this.mount = this.createClientResourceRequest(wow.mount.bind(this, 'static'));
        this.mountSearch = this.createClientResourceRequest(wow.mountSearch.bind(this, 'static'));
        this.mythicKeystone = this.createClientResourceRequest(wow.mythicKeystone.bind(this, 'dynamic'));
        this.mythicKeystoneAffix = this.createClientResourceRequest(wow.mythicKeystoneAffix.bind(this, 'static'));
        this.mythicKeystoneLeaderboard = this.createClientResourceRequest(wow.mythicKeystoneLeaderboard.bind(this, 'dynamic'));
        this.mythicRaidLeaderboard = this.createClientResourceRequest(wow.mythicRaidLeaderboard.bind(this, 'dynamic'));
        this.pet = this.createClientResourceRequest(wow.pet.bind(this, 'static'));
        this.playableClass = this.createClientResourceRequest(wow.playableClass.bind(this, 'static'));
        this.playableRace = this.createClientResourceRequest(wow.playableRace.bind(this, 'static'));
        this.playableSpecialization = this.createClientResourceRequest(wow.playableSpecialization.bind(this, 'static'));
        this.powerType = this.createClientResourceRequest(wow.powerType.bind(this, 'static'));
        this.profession = this.createClientResourceRequest(wow.profession.bind(this, 'static'));
        this.pvpSeason = this.createClientResourceRequest(wow.pvpSeason.bind(this, 'dynamic'));
        this.pvpTier = this.createClientResourceRequest(wow.pvpTier.bind(this, 'static'));
        this.quest = this.createClientResourceRequest(wow.quest.bind(this, 'static'));
        this.realm = this.createClientResourceRequest(wow.realm.bind(this, 'dynamic'));
        this.realmSearch = this.createClientResourceRequest(wow.realmSearch.bind(this, 'dynamic'));
        this.recipe = this.createClientResourceRequest(wow.recipe.bind(this, 'static'));
        this.region = this.createClientResourceRequest(wow.region.bind(this, 'dynamic'));
        this.reputation = this.createClientResourceRequest(wow.reputation.bind(this, 'static'));
        this.soulbind = this.createClientResourceRequest(wow.soulbind.bind(this, 'static'));
        this.spell = this.createClientResourceRequest(wow.spell.bind(this, 'static'));
        this.spellSearch = this.createClientResourceRequest(wow.spellSearch.bind(this, 'static'));
        this.talent = this.createClientResourceRequest(wow.talent.bind(this, 'static'));
        this.techTalent = this.createClientResourceRequest(wow.techTalent.bind(this, 'static'));
        this.title = this.createClientResourceRequest(wow.title.bind(this, 'static'));
        this.token = this.createClientResourceRequest(wow.token.bind(this, 'dynamic'));
    }
}

exports.WoW = WoW;
