// blizzard.js/dist/d3/client.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var core = require('../core/core.js');
var d3 = require('../resources/d3.js');

class D3 extends core.Blizzard {
    constructor() {
        super(...arguments);
        this.act = this.createClientResourceRequest(d3.act);
        this.artisan = this.createClientResourceRequest(d3.artisan);
        this.recipe = this.createClientResourceRequest(d3.recipe);
        this.follower = this.createClientResourceRequest(d3.follower);
        this.characterClass = this.createClientResourceRequest(d3.characterClass);
        this.characterSkill = this.createClientResourceRequest(d3.characterSkill);
        this.item = this.createClientResourceRequest(d3.item);
        this.itemType = this.createClientResourceRequest(d3.itemType);
        this.accountProfile = this.createClientResourceRequest(d3.accountProfile);
        this.season = this.createClientResourceRequest(d3.season);
        this.era = this.createClientResourceRequest(d3.era);
    }
}

exports.D3 = D3;
